### PR TITLE
Minor tweaks to the window scaling code

### PR DESCRIPTION
**Pull request by obilodeau**
Two main changes

### Scrollbar only displayed if replay is not scaled (the default)

It's fixed. Dispatched events needed to be re-ordered because resize was called before QRemoteDesktop's state was updated and I was querying it.

### Few pixels were missing when scaled

Before change:
pyrdp-pixels-grugés

After change:
pyrdp-pixels-ok

The difference between using the viewport() and the height minus scrollbar height is 2 pixels but that was enough to be perceptible.

### CODE REVIEW

File: pyrdp/ui/qt.py [hunk 1]
```diff
@@ -166,6 +166,7 @@ def runOnMainThread(self, target: callable):
     @property
     def screen(self):
         return self._buffer
+
```
Comment (Res260) [inline pyrdp/ui/qt.py:169]: <3

File: pyrdp/ui/qt.py [hunk 2]
```diff
@@ -194,6 +195,12 @@ def scale(self, scale):
         :param scale: Ex: 0.5 for 50% height and 50% width.
         """
         self.ratio = scale
+        if not self.scaleToWindow:
+            # resize widget to session size
+            super().resize(self.sessionWidth, self.sessionHeight)
```
Comment (Res260) [inline pyrdp/ui/qt.py:200]: wouldn't it be more readable using `if condition x else y`  instead of `if not condition y else x`?

Comment (obilodeau) [reply pyrdp/ui/qt.py:200]: Good point

Comment (obilodeau) [reply pyrdp/ui/qt.py]: fixed

**Comment by obilodeau** [comment]
Managed to fix it. Give it a try @Res260.

**Comment by Res260** [comment]
Works well!

**Comment by obilodeau** [comment]
Now that review has happened, I `rebased -i` and squashed fixup commits together. Ready to merge when CI finishes.

**Pull request merged**